doc/source/programs/gdal_cli_from_python.rst

```diff
@@ -0,0 +1,128 @@
+.. _gdal_cli_from_python:
+
+================================================================================
+How to use "gdal" CLI algorithms from Python
+================================================================================
+
+Raster commands
+---------------
+
+* Getting information on a raster dataset as JSON
+
+.. code-block:: python
+
+    from osgeo import gdal
+    import json
+
+    gdal.UseExceptions()
+    alg = gdal.GetGlobalAlgorithmRegistry()["raster"]["info"]
```

Comment:
Accessing an "algorithm registry" still feels a bit low-level to me. What do you think about something like:

```
with gdal.run("raster info", {"input" :"byte.tif"}) as result:
    info = json.loads(result.output)
```

Reply:
An algorithm could have several output arguments (like generating a main raster and a mask). So ``result.output`` would only be allowed if the algorithm declare a single output argument ?

Reply:
> Is this needed at all? A Python user can already do subprocess.run(["raster", "info"])

Maybe they can, though if they're like me and they've only been using Python since 2.1, they still have to look up the documentation for `subprocess` every single time. How do I capture stdout again?

> No need for a context manager.

Not in this example - the idea was to avoid having to call a `Finalize()` method in subsequent examples.